### ISSUE TITLE
Precision

### ISSUE THREAD

**Issue by igneus** [tmlee/time_difference#19]
Recently I was debugging code which relied on TimeDifference when computing difference in days between two `Date`s. It turned out that code like
```ruby
TimeDifference.between(
  Date.new(2015, 3, 15), 
  Date.new(2015, 4, 14)
).in_days.to_i
```
returns on different platforms different results. Having read how TimeDifference makes the calculations (dividing difference in seconds by an approximate length of a year / month / day / ... in seconds) I'm not surprised any more by the results, but for the sake of others tempted to make the same mistake I suggest being more explicit in the README concerning accuracy: TimeDifference is great for quick calculations where no accuracy is required (how long is it since I wrote my blog post?), but unsuitable for accurate computations (for how many days of using the service will I be billed?).

**Comment by joshuapinter** [comment]
Yes, I noticed this as well and it was causing small but significant (to our overall calculations and logic) issues. Here is the example we ran into: 

```ruby
start_at
#=> Thu, 17 Apr 2025 15:16:15.097000000 CDT -05:00

end_at
#=> Thu, 17 Apr 2025 15:16:29.360500097 CDT -05:00

TimeDifference.between( start_at, end_at ).in_hours
#=> 0.0

( end_at - start_at ) / 1.hour
#=> 0.003962083360354106
```

As you can see, `TimeDifference` returns a value of 0.0 hours for about a 15 second gap instead of showing the actual value, which is around 0.004.

This caused cascading `NaN` issues down our workflow because we ended up with denominators of 0.0 instead of 0.003962083360354106.

I suppose the best way forward is to replace all uses of `TimeDifference` with the less clean version of `( end - start ) / 1.hour` or whatever you're looking for.
